### DISCUSSION TITLE
ufw阻止natter.py打洞

### DISCUSSION THREAD

**Discussion by XiaoTong6666** [MikeWang000000/Natter#63] [General]
一开始我的电脑使用的是Win(nt6.3)系统，防火墙正常打开，路由DMZ和UPnP都打开，洞也正常能打，使用--check-nat测试出来是NAT1，正常使用了一段时间。
最近我重装系统，换成更轻量的Ubuntu(Linux6.5.0-17)在迁移服务的时候，发现这个natter怎么测都是NAT3，起初以为是运营商的问题，找装维说是NAT1我也重置了路由器，光猫，最终电脑拨号都以失败告终
在刚刚就想，有没有一种可能，就是防火墙的问题，接着我就把ufw关了，再用--check-nat测试，变成NAT1了
然后现在我不知道应该怎么修改ufw的配置文件。关掉ufw也不行，但是又得使用natter打洞。所以大佬们可否给我指一条路，给点指示

**Comment by XiaoTong6666** [comment]
![IMG_20240221_041244](https://github.com/MikeWang000000/Natter/assets/110387028/02e9a2bc-b853-4dad-83fe-d034aeeb4575)

**Comment by MikeWang000000** [comment]
关于防火墙，我很难给出好的建议。

因为Natter默认选择的端口范围是动态的，这个范围来自 `/proc/sys/net/ipv4/ip_local_port_range` 。

由于您的机器处于内网，可以考虑放行 IPv4 这个端口范围内的所有 TCP / UDP 传入流量。

或者使用 2.0 的 `-b` 选项，绑定单个端口，这样仅需放行单个端口。

**Comment by XiaoTong6666** [comment]
好的，感谢大佬的解答。

**Reply by fishheadhead** [reply]
你好，我也使用的ufw管理防火墙，遇到了和你同样的问题。作者建议使用-b参数，我关闭防火墙，绑定测试了多个端口，没有办法运行，请问你是最后用的什么方案呢？
